tests/lib/rules/no-else-return.js

```diff
@@ -182,6 +182,32 @@ ruleTester.run("no-else-return", rule, {
             output: "function foo21() { var x = true; if (x) { return x; } if (x === false) { return false; } }",
             options: [{ allowElseIf: false }],
             errors: [{ messageId: "unexpected", type: "IfStatement" }]
+        },
+        {
+            code: "function foo() { if (true) { return bar; } else { const baz = 1;  return baz; } }",
+            output: null,
+            parserOptions: { ecmaVersion: 6 },
+            errors: [{ messageId: "unexpected", type: "BlockStatement" }]
+        },
+        {
+            code: "function foo() { if (true) { return bar; } else { let baz = 1;  return baz; } }",
+            output: null,
```

Comment:
It might be okay to just not autofix any block scope declarations to be safe, but I think this one *could* be autofixed since none of the declarations share a name with the variable in the immediate upper scope.

Reply:
Hi @platinumazure , I'm sorry for the delay. I have started to work with this issue.
What the expected output here ?